File: solutions/swift/raindrops/3/Sources/Raindrops/Raindrops.swift
private let raindropRules = [(3, "Pling"), (5, "Plang"), (7, "Plong")]

func raindrops(_ number: Int) -> String {
    let result = raindropRules
        .compactMap { number.isMultiple(of: $0) ? $1 : nil }
        .joined()

    return result.isEmpty ? String(number) : result
}

File: solutions/elixir/raindrops/5/lib/raindrops.ex
defmodule Raindrops do

  defp number_to_drop do
    [
      &if(rem(&1, 3) == 0, do: "Pling"),
      &if(rem(&1, 5) == 0, do: "Plang"),
      &if(rem(&1, 7) == 0, do: "Plong")
    ]
  end

  def convert(number) do
    drops = for splash_pattern <- number_to_drop(),
            drop = splash_pattern.(number),
            drop != nil,  do: drop

    case drops do
      [] -> to_string(number)
      _  -> Enum.join(drops)
    end
  end

end
